server/src/languageServerBase.ts

```diff
@@ -805,43 +768,12 @@ export class LanguageServerBase {
         });
     }
 
-    private convertRange(range?: DiagnosticTextRange): Range {
-        if (!range) {
-            return Range.create(this.convertPosition(), this.convertPosition());
-        }
-        return Range.create(this.convertPosition(range.start), this.convertPosition(range.end));
-    }
-
-    private convertPosition(position?: DiagnosticTextPosition): Position {
-        if (!position) {
-            return Position.create(0, 0);
-        }
-        return Position.create(position.line, position.column);
-    }
-
-    private convertUriToPath(uriString: string): string {
```

Comment:
If you're replacing the implementation of convertUriToPath, please make sure to exercise the functionality on a Windows machine where paths contain DOS-style paths. I don't currently have unit tests in place for this case because. (Yes, I really should.)

Reply:
Sure, working on it. BTW, stub creation fails if `typingsPath` is not specified, although doc says default is `./typings`. I can make it `projectRoot/typings` if you want.

Reply:
Looks like `typingsPath` is set to default only if `pyrightconfig.json` exists. Otherwise config watcher is not created and settings are not loaded from JSON (and that's where the default is set).

Reply:
Looks like unrelated existing issues. Filed https://github.com/microsoft/pyright/issues/494.